src/service/rate/rates.ts

```diff
@@ -0,0 +1,48 @@
+import { Container, Service } from "typedi";
+import { TatumConnector } from "../../connector/tatum.connector";
+import { CONFIG, ErrorUtils, ResponseDto } from "../../util";
+import { Rate, RateBatchDto } from "./rates.dto";
+import { ApiVersion, TatumConfig } from "../tatum";
+
+@Service({
+  factory: (data: { id: string }) => {
+    return new Rates(data.id);
+  },
+  transient: true
+})
+export class Rates {
+  private connector: TatumConnector;
+  private readonly config: TatumConfig
+
+  constructor(private readonly id: string) {
+    this.config = Container.of(this.id).get(CONFIG)
+    this.connector = Container.of(this.id).get(TatumConnector)
+  }
+
+  getCurrentRate(currency: string, basePair: string): Promise<ResponseDto<Rate>> {
+    return ErrorUtils.tryFail(async () => {
+      return this.connector.get<Rate>({
+        path: `${this.getUrlPrefix()}/${currency}`,
+        params: {basePair}
+      });
+    });
+  }
+
+  getCurrentRateBatch(pairs: RateBatchDto[]): Promise<ResponseDto<Rate>> {
+    pairs.forEach(pair => {
+      if (!pair.batchId) {
+        pair.batchId = `${pair.currency}/${pair.basePair}`;
+      }
+    })
+    return ErrorUtils.tryFail(async () => {
+      return this.connector.post<Rate>({
+        path: `${this.getUrlPrefix()}`,
+        body: pairs
+      });
+    });
+  }
+
+  private getUrlPrefix() {
+    return this.config.version === ApiVersion.V1 ? 'tatum/rate' : 'rate'
```

Comment:
we should support only new ApiVersion.V2 here - not fallback to V1 pls

Reply:
I think we should directly go to the v2, because there is no v1.